client/my-sites/controller.js

```diff
@@ -109,7 +109,7 @@ module.exports = {
 	 * Set up site selection based on last URL param and/or handle no-sites error cases
 	 */
 	siteSelection( context, next ) {
-		const siteID = route.getSiteFragment( context.path );
+		const siteID = context.params.site_id;
```

Comment:
I'm a little confused about this change. What effect does it have, and could it affect other parts of site-settings? Is it used in this PR?